src/main.cpp

```diff
@@ -13,55 +13,219 @@
+    }
+
+    // Process model files
+    for (const auto& model_path : model_paths) {
+        fs::path base_path = fs::path(model_path);
+
+        // Only process .mdl files
+        if (base_path.extension() != ".mdl") {
+            continue;
+        }
+
+        fs::path dst_base_path = models_output_dir / base_path;
+        fs::create_directories(dst_base_path.parent_path());
+
+        // Copy all related files first
+        for (const auto& ext : extensions) {
+            fs::path current_path = base_path;
+            current_path.replace_extension(ext);
+            std::string current_path_str = current_path.string();
+
+            fs::path dst_path = dst_base_path;
+            dst_path.replace_extension(ext);
+
+            copy_file_from_vpk(current_path_str, dst_path);
+        }
+
+        // Now process the .mdl file with rmdlconv
+        fs::path mdl_path = dst_base_path;
+        fs::path rmdlconv_out_dir = dst_base_path.parent_path() / "rmdlconv_out";
+
+        // Prepare command line
+        std::wstring cmdLine = L"\"" + rmdlconv_path.wstring() + L"\" -targetversion 53 -nopause -convertmodel \"" + mdl_path.wstring() + L"\"";
+
+        // Set up process information
+        STARTUPINFOW si = { sizeof(si) };
+        PROCESS_INFORMATION pi;
+
+        // Create the process
+        if (CreateProcessW(NULL, &cmdLine[0], NULL, NULL, FALSE, 0, NULL, NULL, &si, &pi))
+        {
+            // Wait for the process to finish
+            WaitForSingleObject(pi.hProcess, INFINITE);
+
+            // Get the exit code
+            DWORD exitCode;
+            GetExitCodeProcess(pi.hProcess, &exitCode);
+
+            // Close process and thread handles
+            CloseHandle(pi.hProcess);
+            CloseHandle(pi.hThread);
+
+            if (exitCode == 0) {
+                fs::path converted_mdl = rmdlconv_out_dir / (mdl_path.filename().string() + "_new");
+                if (fs::exists(converted_mdl)) {
+                    // Remove original files
+                    for (const auto& ext : extensions) {
+                        fs::path to_remove = dst_base_path;
+                        to_remove.replace_extension(ext);
+                        fs::remove(to_remove);
+                    }
+                    // Move converted file
+                    fs::rename(converted_mdl, mdl_path);
+                    printf("Converted and replaced: %s\n", mdl_path.string().c_str());
+                }
+            }
+        }
+        else {
+            printf("Failed to start rmdlconv process. Error code: %d\n", GetLastError());
+        }
+
+        // Clean up rmdlconv_out directory
+        fs::remove_all(rmdlconv_out_dir);
+    }
 }
 
 
 int main(int argc, char* argv[]) {
-    if (argc != 3) {
+    if (argc != 4) {
         print_usage(argv[0]);
         return 0;
     }
-    char* in_filename  = argv[1];
-    char* out_filename = argv[2];
+    char* game_dir = argv[1];
```

Comment:
since we've changed to pulling the map directly from the `.vpk`, the old method is gone?
what about Titanfall: Online maps? they don't come in `.vpk`s

Reply:
I'm ok with this being a thing for this version, since we can use v1.0.0 for TF:O maps
For now.
After #12 is resolved we'll have a problem

Reply:
having an options flag to switch between modes would be nice (e.g. `--from-vpks`)
but that would definitely complicate parsing args

Reply:
you also need to update the `Usage` section of `README.md` to reflect the new method

personally I'd like to use something like this:
```
./bsp_regen/
  | bsp_regen.exe
  | r1/
  | | maps/
  | \ models/
  \ r2/
    | maps/
    \ models/
```
where users can put maps & models to convert in a local `r1/` folder
then converted maps can be collected from `r2/`
any batching can be done with a script that calls `bsp_regen.exe`

I'd just like to maintain some compatibility with the non-vpk method